app/src/main/java/com/sopt/now/compose/presentation/SignUp/SignUpViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package com.sopt.now.compose.presentation.SignUp
+
+
+import android.content.ContentValues.TAG
+import android.util.Log
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sopt.now.compose.domain.AuthRepository
+import com.sopt.now.compose.domain.model.AuthEntity
+import kotlinx.coroutines.launch
+import retrofit2.HttpException
+class SignUpViewModel(
+    private val authRepository: AuthRepository
+) : ViewModel() {
+    val signUpResult: MutableLiveData<Boolean> = MutableLiveData()
+    val errorMessage: MutableLiveData<String> = MutableLiveData()
+    val userId: MutableLiveData<String> = MutableLiveData()
+
+    var signUpId = mutableStateOf("")
+    var signUpPw = mutableStateOf("")
+    var signUpName = mutableStateOf("")
+    var signUpPhone = mutableStateOf("")
```

Comment:
```suggestion
   var signUpId by remember { mutableStateOf("") }
   var signUpPw by remember { mutableStateOf("") }
   var signUpName by remember { mutableStateOf("") }
   var signUpPhone by remember { mutableStateOf("") }
```
이렇게 해야 리컴포지션 영향을 받지 않을 것 같아요!

remember는 상태를 유지하여 리컴포지션이 발생하더라도 입력된 값이 초기화되지 않도록 해준다고 하네요!
by remember를 같이 사용하면 상태관리에 더욱 효율적일 것 같아요!

Reply:
https://github.com/NOW-SOPT-ANDROID/chanwoo-bae/pull/17#discussion_r1590347708

```kotlin
var id by rememberSaveable { mutableStateOf("") }
var password by rememberSaveable { mutableStateOf("") }
var nickName by rememberSaveable { mutableStateOf("") }
var mbti by rememberSaveable { mutableStateOf("") }
```

rememberSaveable: 컴포넌트의 생명주기 동안 상태를 유지하며, 프로세스 종료 및 구성 변경(예: 화면 회전) 시에도 상태를 유지합니다.

이런것도 있다고 합니다!!